main.cpp

```diff
@@ -28,510 +28,592 @@ Thing: Car Wash
     void recogniseFace(bool awake, int numberMeetingsWithPerson);
-//    2) estimate distances
-    void estimateDistance(Vision vision);
-//    3) construct a language
+    void estimateDistance(int distance);
     void constructLanguage(int age);
 };
 
-/*
-Thing 4) chord
-5 properties:
-    1) number of notes
-    2) possible next chords within key
-    3) chord number in roman numerals
-    4) has chord been altered
-    5) possible preceding chords within key
-3 things it can do:
-    1) make a sound
-    2) be played as arpeggio
-    3) be omitted
- */
+void Brain::recogniseFace(bool awake, int numberMeetingsWithPerson)
+{
+    if (awake && numberMeetingsWithPerson >= 2)
+    {
+        std::cout << "Face recognised" << std::endl;
+    }
+    else
+    {
+        std::cout << "Face not recognised" << std::endl;
+    }
 
-struct Chord
+}
+
+void Brain::estimateDistance(int distance)
 {
+    if (distance < 250)
+    {
+        std::cout << "Distance estimated" << std::endl;
+    }
+    else
+    {
+        std::cout << "Distance could not be estimated" << std::endl;
+    }
+
+
+}
 
-//    1) number of notes
+void Brain::constructLanguage(int age)
+{
+    bool languageSuccess;
+    if (age >= 2)
+    {
+        languageSuccess = true;
+    }
+    else
+    {
+        languageSuccess = false;
+    }
+}
+void Brain::Vision::checkVision(double eyesightQuality)
+{
+    if (eyesightQuality < 45.8)
+    {
+        getGlassesPrescription();
+    }
+}
+
+void Brain::Vision::probabilityOfSighting(bool morning)
+{
+    if (morning)
+    {
+        std::cout << " >=50% Chance of sighting" << std::endl;
+    }
+    else
+    {
+        std::cout << "<50% chance of sighting" << std::endl;
+    }
+
+}
+
+void Brain::Vision::getGlassesPrescription()
+{
+    std::cout << "prescription needed" << std::endl;
+}
+
+struct Chord
+{
     int numberOfNotes = 4;
-//    2) possible next chords within key
     std::string nextChordList = "F#7, B";
-//    3) chord number in roman numerals
     std::string chordInRomanNumerals = "V7";
-//    4) has chord been altered
     bool alteredChord = false;
-//    5) possible preceding chords within key
     std::string precedingChordList = "C#m, D#m";
 
-//    1) make a sound
-    void makeSound(int NoteNumber);
-//    2) be played as arpeggio
+    int makeSound(int noteNumber);
     int playArpeggio(int numberOfHeldNotes);
-//    3) be omitted
-    void playNothing(std::string chord);
+    int playNothing(int numberOfNotes);
 };
 
-/*
-Thing 5) Keys
-5 properties:
-    1) key depressed
-    2) velocity of key pressed (1-127)
-    3) MIDI number of note pressed (1-127)
-    4) note legato value (note held, while another is played)
-    5) keyboard set to monophonic
-3 things it can do:
-    1) play legato
-    2) play chords
-    3) play monophonic
- */
+int Chord::makeSound(int noteNumber)
+{
+    if (noteNumber)
+    {
+        return noteNumber;
+    }
+    else
+    {
+        return playNothing(0);
+    }
```

Comment:
If you find yourself writing code like this:
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    else 
    {
        return anotherValue;
    }
}
```
if the `if()` block is entered, the `return aValue;` is the last line of code to be executed.
you have no code after the `else` block that needs executing, since your `else` block contains a `return` statement.
`return` statements are the last thing that **ever happens** in a function
That means if your `if()` block is **NOT** entered, it will be skipped.  What is the next thing that happens?  `return anotherValue;`
that means you do **NOT** need the else block, and it is just causing visual clutter.
```
int Type::someFunction()
{
    if( someExpression )
    {
        someStatement();
        return aValue;
    } 
    return anotherValue;
}
```
- [ ] revise

Reply:
Done.